static/src/components/setting/quiz_setting.vue

```diff
@@ -0,0 +1,293 @@
+              >
+                Reset
+              </button>
+            </td>
+          </tr>
+        </tbody>
+      </table>
+      <div>
+        <button
+          class="btn btn-sm btn-primary"
+          @click="save"
+        >
+          Save
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+
+  props: {
+    csrfToken: {
+      type: String,
+      default: null
+    },
+    config: {
+      type: Object,
+      default: () => ({ enabled: false, questions: 0, passing: 0, complete_mode: null, short_circuit: false, mode_choices: [] })
+    },
+    allUserQuiz: {
+      type: Array,
+      default: function () { return []; }
+    }
+  },
+
+  data () {
+    return {
+      enabled: this.config.enabled,
+      questions: this.config.questions,
+      passing: this.config.passing,
+      mode: this.config.completion_mode,
+      resetUser: [],
+      users: {}
+    };
+  },
+
+  mounted: function () {
+    this.users = this.getQuizUser();
+  },
+
+  methods: {
+
+    getOptions () {
+      var options = [];
+      var i;
+      for (i = 0; i < this.config.mode_choices.length; i++) {
+        var opt = {
+          text: this.config.mode_choices[i][1],
+          value: this.config.mode_choices[i][0]
+        };
+        options.push(opt);
+      }
+      return options;
+    },
+
+    getQuizUser () {
+      var users = {};
+      for (var i = 0; i < this.allUserQuiz.length; i++) {
+        var u = this.allUserQuiz[i];
+        if (u.quiz.status === 'bg-success') {
+          u.quiz.status = 'pass';
+        } else if (u.quiz.status === 'bg-danger') {
+          u.quiz.status = 'fail';
+        }
+        users[u.id] = u;
+      }
+      return users;
+    },
+
+   reset (event, id) {
+     this.resetUser.push(id);
+   },
+
+    _isIntegerNumeric: function (_n) {
+        return Math.floor(_n) === _n;
+    },
+
+    async save () {
+      const _questions = parseInt(this.questions);
+      const _passing = parseInt(this.passing);
+      if (!this._isIntegerNumeric(_questions) || !this._isIntegerNumeric(_passing)) {
+        return;
+      }
+      try {
+        const res = await fetch(window.location.pathname, {
+          method: 'POST',
+          headers: {
+            'content-type': 'application/json',
+            'X-CSRFToken': this.csrfToken
+          },
+          credentials: 'same-origin',
+          body: JSON.stringify({ 'quiz': {
+
+              enabled: this.enabled,
+              questions: _questions,
+              passing: _passing,
+              completion_mode: this.mode,
+
+            reset: this.resetUser
+          } })
+        });
+        if (res.ok) {
+          window.pybossaNotify('quiz data Saved.', true, 'success');
+        } else {
+          window.pybossaNotify('An error occurred.', true, 'error');
+        }
+      } catch (error) {
+        window.pybossaNotify('An error occurred.', true, 'error');
+      }
+     }
+  }
+};
+
+</script>
```

Comment:
I think it is a decent practice to scope a component style by default, see for example 

https://github.com/bloomberg/pybossa-default-theme/blob/fd0ed64026aa7237a0447f5b58647f65b4474fa3/static/src/components/builder/components/TextInput/TextInputForm.vue#L51-L57